rapid_response_xblock/static/js/src/rapid.js

```diff
@@ -427,21 +448,60 @@
 
       newContainers.merge(containers)
         .attr("class", "chart-container " + (chartKeys.length === 1 ? 'single-chart' : 'two-charts'))
-        .each(function (index, __, charts) {
-          renderChart(d3.select(charts[index]), index);
-        })
         .selectAll("svg")
         .attr("width", calcChartWidth())
         .attr("height", calcChartHeight())
         .selectAll(".chart")
         .attr("transform", "translate(" + ChartSettings.left + "," + ChartSettings.top + ")")
         .select(".xaxis");
 
+      newContainers.each(function (_, index, charts) {
+        renderChart(d3.select(charts[index]), index);
+      });
+
       // Remove charts if selectedRuns reduces in size.
       // We don't need to do this for all the inner elements, the remove will propagate.
       containers.exit().remove();
     }
 
+    function renderMathHJax(chart) {
+      setTimeout(function() {
```

Comment:
Can you explain why this timeout and the timeout at line 476 are necessary? If we really need them, can you move this particular `setTimeout` to where `renderMathHJax` is called? Would look like this:

```javascript
      //...
      chart.select(".yaxis .line")
        .transition()	        .transition()
        .attr("y2", innerHeight);	        .attr("y2", innerHeight);

      setTimeout(function() {
        renderMathHJax(chart);
      }, 1);
```

Reply:
it turned out we dont need outer `setTimeout`. I was just following example

- https://stackoverflow.com/questions/37359888/mathjax-not-rendering-inline-labels
- http://bl.ocks.org/larsenmtl/86077bddc91c3de8d3db6a53216b2f47